### ISSUE TITLE
Prevent misleading search results

### ISSUE THREAD

**Issue by adambard1** [streetlives/yourpeer.nyc#119]
- [ ] Stop misleading search results from appearing high up the search result list

Currently, search results that don't contain the searched term appear high up the results list.
This is important to fix since those unfamiliar with locations will be directed to incorrect locations

e.g. If I search "streetwork" (the location name of one of the city's most popular service locations) one correct result heads the list, but two incorrect results appear before the next correct result
The two incorrect results don't appear to contain the search term or any part of the search term in any field.

<img width="488" alt="Screenshot 2024-05-27 at 10 51 56" src="https://github.com/streetlives/yourpeer.nyc/assets/13530245/1cdaf975-4bd7-4cc6-a2c3-9e56c8ac6ffb">

**Comment by doobneek1** [comment]
![Screenshot 2024-05-16 203907](https://github.com/streetlives/yourpeer.nyc/assets/170349995/91b1d70f-dc3c-4cf6-9696-926965aa14f2)
Again, the search shows borowide irrelevant results and the actual clinic is not the first result

**Comment by doobneek1** [comment]
https://yourpeer.nyc/locations?search=weil+cornell
the most relevant result is not first here, I think the hierarchy of where the match is found should be enforced. 
![image](https://github.com/streetlives/yourpeer.nyc/assets/170349995/9fa93596-5180-425d-b554-2032499f00ec)

**Comment by adambard1** [comment]
yes, that's meant to be the case - exact matching, with org matching first, location matching second, etc.

**Comment by doobneek1** [comment]
https://yourpeer.nyc/locations?search=brooklyn+ghost
![image](https://github.com/streetlives/yourpeer.nyc/assets/170349995/56e27cc8-bd4c-4251-b0a4-190a8b81ffb5)
the most relevant search result pops up on the left first but the map shows a bunch of irrelevant results

**Comment by doobneek1** [comment]
https://yourpeer.nyc/locations?search=fair+fares
A lot of irrelevant results here
